openapi-core/src/main/java/io/ballerina/openapi/core/generators/service/ServiceDiagnosticMessages.java

```diff
@@ -49,7 +49,10 @@ public enum ServiceDiagnosticMessages {
             "Header '%s' with no header type can not be mapped to the Ballerina headers.",
             DiagnosticSeverity.ERROR),
     OAS_SERVICE_107("OAS_SERVICE_107", "HTTP status code '%s' is not supported in Ballerina.",
-            DiagnosticSeverity.ERROR);
+            DiagnosticSeverity.ERROR),
+
+    OAS_SERVICE_108("OAS_SERVICE_108", "Header '%s' can not be mapped as a valid Ballerina header parameter.",
+                    DiagnosticSeverity.ERROR);
```

Comment:
Shouldn't we be specific about the reason in here? 

How about `Header 'foo' with type: 'bar' is not supported in Ballerina.`?

Reply:
Updated to `Header '%s' with type '%s' can not be mapped as a valid Ballerina header parameter.`